### PR TITLE
Add request/responce types for listVersions API

### PR DESCRIPTION
**Pull request by LiubovDidkivska**
Add Versions request and responce types.
Added VersionDependency, VersionInfo and VersionInfos types, which will be used in List versions API.

Relates-To: OLPEDGE-1606

Signed-off-by: Liubov Didkivska <ext-liubov.didkivska@here.com>

### CODE REVIEW

File: olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/VersionsRequest.h [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+   * @see `GetFetchOption()` for information on usage and format.
+   *
+   * @param fetch_option The `FetchOption` enum.
+   *
+   * @return A reference to the updated `VersionsRequest` instance.
+   */
+  inline VersionsRequest& WithFetchOption(FetchOptions fetch_option) {
+    fetch_option_ = fetch_option;
+    return *this;
+  }
```
Comment (mykhailo-kuchma) [inline olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/VersionsRequest.h:162]: @andescu, what is your opinion? Do we need that?

File: olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/model/Versions.h [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+ * @brief Represents a catalog version
+ */
+class DATASERVICE_READ_API CatalogVersion final {
+ public:
+  /**
+   * @brief Sets the vector of catalog version dependencies.
+   *
+   * @param vector of catalog version dependencies.
+   */
+  void SetCatalogVersionDependencies(
```
Comment (andescu) [inline olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/model/Versions.h:98]: Please follow naming from spec.
There you have the following fields:
- dependencies
- partitionCounts
- timestamp
- version

Means the getters/setters need to have the same name:
- `GetDependencies` / `SetDependecies`
- `GetPartitionCounts` / `SetPartitionCounts`
and so on.

File: olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/model/Versions.h [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+ private:
+  int64_t version_;
+  int64_t timestamp_;
+  std::vector<CatalogVersionDependencies> dependencies_;
+  std::map<std::string, int64_t> partition_counts_;
+};
+/**
+ * @brief A container for list of versions.
+ */
+class DATASERVICE_READ_API Versions final {
```
Comment (andescu) [inline olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/model/Versions.h:168]: Move to own file.

File: olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/VersionsRequest.h [hunk 4]
```diff
@@ -68,10 +68,10 @@ class DATASERVICE_READ_API VersionsRequest final {
...
    * @brief Sets the catalog metadata end version.
    *
-   * @note The end of the range of versions you want to get (inclusive). This
-   * must be a valid catalog version greater than the startVersion. The maximum
-   * value for this parameter is returned from the /versions/latest endpoint. If
-   * this version does not exist, 400 Bad Request is returned.
+   * @note The end of the range of versions that you want to get (inclusive). It
+   * must be a valid catalog version greater than the `startVersion`. The
+   * maximum value for this parameter is returned from the `/versions/latest`
+   * endpoint. If this version does not exist, 400 Bad Request is returned..
```
Comment (HalynaDumych) [inline olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/VersionsRequest.h:74]:
```suggestion
   * endpoint. If this version does not exist, 400 Bad Request is returned.
```

File: olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/model/VersionInfo.h [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+
+  /**
+   * @brief Gets the catalog version.
+   *
+   * @return The catalog version.
+   */
+  std::int64_t GetVersion() const { return version_; }
+
+  /**
+   * @brief Gets the timestamp of catalog version.
```
[the remark this answers is not in the document]
Comment (HalynaDumych) [reply olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/model/VersionInfo.h]: Please also commit this suggestion.

File: olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/VersionsRequest.h [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+#include <olp/dataservice/read/DataServiceReadApi.h>
+#include <olp/dataservice/read/FetchOptions.h>
+#include <boost/optional.hpp>
+
+namespace olp {
+namespace dataservice {
+namespace read {
+
+/**
+ * @brief  Encapsulates the fields required to request a list of versions for
```
Comment (HalynaDumych) [inline olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/VersionsRequest.h:36]:
```suggestion
 * @brief Encapsulates the fields required to request a list of versions for
```

File: olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/model/VersionInfo.h [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+
+  /**
+   * @brief Sets the catalog version.
+   *
+   * @param The catalog version.
+   */
+  void SetVersion(std::int64_t version) { version_ = version; }
+
+  /**
+   * @brief  Sets the timestamp of the catalog version.
```
Comment (HalynaDumych) [inline olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/model/VersionInfo.h:68]:
```suggestion
   * @brief Sets the timestamp of the catalog version.
```

File: olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/model/VersionInfos.h [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+class DATASERVICE_READ_API VersionInfos final {
+ public:
+  /**
+   * @brief Sets the vector of version infos.
+   *
+   * @param value The vector of version infos .
+   */
+  void SetVersionInfos(std::vector<VersionInfo> value) {
+    versions_ = std::move(value);
+  }
```
Comment (mykhailo-kuchma) [inline olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/model/VersionInfos.h:48]: SetVersions

File: olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/model/VersionInfos.h [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+
+  /**
+   * @brief Gets the vector of version infos.
+   *
+   * @return The vector of version infos.
+   */
+  const std::vector<VersionInfo>& GetVersionInfos() const { return versions_; }
```
Comment (mykhailo-kuchma) [inline olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/model/VersionInfos.h:55]: GetVersions

File: olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/model/VersionInfo.h [hunk 10]
```diff
@@ -0,0 +1,116 @@
...
+ * @brief Represents a version info.
+ */
+class DATASERVICE_READ_API VersionInfo final {
+ public:
+  /**
+   * @brief Sets the vector of version dependencies.
+   *
+   * @param The vector of version dependencies.
+   */
+  void SetVersionDependencies(std::vector<VersionDependency> dependencies) {
```
Comment (mykhailo-kuchma) [inline olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/model/VersionInfo.h:46]: SetDependencies

File: olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/model/VersionInfo.h [hunk 11]
```diff
@@ -0,0 +1,116 @@
...
+   * @param The timestamp of the catalog version.
+   */
+  void SetTimestamp(std::int64_t timestamp) { timestamp_ = timestamp; }
+
+  /**
+   * @brief Gets the vector of version dependencies.
+   *
+   * @return The vector of version dependencies.
+   */
+  const std::vector<VersionDependency>& GetVersionDependencies() const {
```
Comment (mykhailo-kuchma) [inline olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/model/VersionInfo.h:79]: GetDependencies

**Comment by mykhailo-kuchma** [comment]
Please rename getters and setters according to openapi spec.

File: olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/model/VersionInfos.h (hunk 8 above)
Comment (mykhailo-kuchma) [reply olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/model/VersionInfos.h:48]: Rename value to versions

**Pull request merged**